File: python/30-39/problem36.py
def is_palindrome(x):
    '''Returns a boolean on wether a number or string is a palindrome.

    Args:
        x: Number or string

    Returns:
        Boolean
    '''
    x = str(x)
    return x == x[::-1]


def binary(n):
    '''Changes n (in base 10) to base 2

    Args:
        n: Number

    Return:
        'n' in binary.
    '''
    return int(bin(n)[2:])


def double_base_palindromes(max):
    '''Retuns a list of every number that is a palindrome in both base 10 and base 2.

    Args:
        max: The max number to search for double base palindromes.

    Returns:
        A list of every double base palindrome from 1 to max.
    '''
    return [x for x in range(max + 1) if is_palindrome(x) and is_palindrome(binary(x))]


def main():
    print(sum(double_base_palindromes(1000000)))


if __name__ == '__main__':
    main()

File: js/30-39/problem36.js
const prototypes = require('../prototypes');

var binary = function(dec) {
    return (dec >>> 0).toString(2);
};

var reverseNum = function(n) {
    return parseInt(n.toString().split('').reverse().join(''));
};

var doubleBasePalindromes = function(max) {
    var answer = [];
    for (let i = 0; i < max; i++) {
        if (i === reverseNum(i) && parseInt(binary(i)) === reverseNum(binary(i))) {
            answer.push(i);
        }
    }
    return answer.sum();
};

console.log(doubleBasePalindromes(1000000));
